### PR TITLE
Update links

### PR DESCRIPTION
**Pull request by sunpoet**
# Description:
Update links which are https:// now.

### CODE REVIEW

File: test/rubygems/test_gem_server.rb [hunk 1]
```diff
@@ -473,7 +473,7 @@ def test_valid_homepage_http
     dir = "#{@gemhome}2"
 
     spec = util_spec 'validhomepagegemhttp', 1
-    spec.homepage = "http://rubygems.org"
+    spec.homepage = "https://rubygems.org/"
```
Comment (bronzdoc) [inline test/rubygems/test_gem_server.rb]: Please revert this change, this test is testing for `http` specifically

File: Rakefile [hunk 2]
```diff
@@ -340,7 +340,7 @@ SHA256 Checksums:
 
 #{checksums}
 
-[download]: http://rubygems.org/pages/download
+[download]: https://rubygems.org//pages/download
```
Comment (bronzdoc) [inline Rakefile]: The path has an extra `//`

File: test/rubygems/test_gem_commands_sources_command.rb [hunk 3]
```diff
@@ -247,7 +247,7 @@ def test_execute_add_redundant_source_trailing_slash
   end
 
   def test_execute_add_http_rubygems_org
-    http_rubygems_org = 'http://rubygems.org'
+    http_rubygems_org = 'https://rubygems.org/'
```
Comment (bronzdoc) [inline test/rubygems/test_gem_commands_sources_command.rb]: The test is `test_execute_add_http_rubygems_org` please add a new test for `https` - `test_execute_add_https_rubygems_org`

**Comment by bronzdoc** [comment]
@bundlerbot r+

**Comment by sunpoet** [comment]
@bronzdoc Thanks for your help and review!

File: lib/rubygems/commands/sources_command.rb [hunk 4]
```diff
@@ -136,7 +136,7 @@ def description # :nodoc:
 its history:
 
 * http://gems.rubyforge.org (RubyGems 1.3.6 and earlier)
-* http://rubygems.org       (RubyGems 1.3.7 through 1.8.25)
+* https://rubygems.org/       (RubyGems 1.3.7 through 1.8.25)
```
Comment (tnir) [inline lib/rubygems/commands/sources_command.rb:139]: This line was incorrect. Will be fixed in #5723
